### PR TITLE
feat(arithmetic): Adding a parser for arithmetic

### PR DESCRIPTION
**Pull request by wmak**
- This is just a simple arithmetic parser that can handle just numbers
  for now
- This is so that later it can be extended to handle variables to create
  more complicated Discover columns

### CODE REVIEW

**Review by Zylphrex — APPROVED** [review]
Just a small comment. But otherwise, LGTM

File: src/sentry/discover/arithmetic.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+maybe_factor         = (factor / primary)
+
+factor               = spaces primary remaining_muls
+remaining_muls       = mul_div+
+mul_div              = mul_div_operator primary spaces
+
+add_sub_operator     = spaces (plus / minus) spaces
+mul_div_operator     = spaces (multiply / divide) spaces
+# TODO(wmak) allow variables
+primary              = numeric_value
```
[the remark this answers is not in the document]
Comment (wmak) [reply src/sentry/discover/arithmetic.py]: :+1: good idea, made the visitors for all three the same too which is nice

**Review by shruthilayaj — APPROVED** [review]

**Review by k-fish — APPROVED** [review]
Arithmetic 😮 .

What happens with nested parens and order of operations? Is there a test for it? (should there be?)

**Comment by wmak** [comment]
@k-fish so this should have basic order of operations (ie. multiplication & division have higher priortity than addition and subtraction) but not brackets **yet** (ie. DMAS but not BEDMAS) and is currently roughly tested by
https://github.com/getsentry/sentry/pull/26057/files#diff-a56d3d53717fef131973ee39a0a7a85999414265d37573db0a137b381cc4a3dbR83-R91

I'll add a few more, its just that I can't do the same amount of parametrization as the other tests since i have to craft the asserts by hand

**Pull request merged**